src/objective/multiclass_objective.hpp

```diff
@@ -46,11 +60,12 @@ class MulticlassLogloss: public ObjectiveFunction {
           auto p = rec[k];
           size_t idx = static_cast<size_t>(num_data_) * k + i;
           if (label_int_[i] == k) {
-            gradients[idx] = static_cast<score_t>(p - 1.0f);
+            gradients[idx] = static_cast<score_t>(p - 1.0f) * label_pos_weights_[k];
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p))* label_pos_weights_[k];
           } else {
             gradients[idx] = static_cast<score_t>(p);
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p));
```

Comment:
We can try different heuristics. I was thinking of using the loss itself as the departure point, and from there derive what the hessian and gradient should be like. I would be happy to try both.

Now I am a bit busy with other things, but the loss-based one I discussed about is quick to implement. I will try to make some time today or early next week.